float64 maxSpeed
float64 acceleration